File: ruby/lib/rock_paper_scissors.rb
class Game

  PAIRS = {
    rock:     { beats: :scissors },
    paper:    { beats: :rock },
    scissors: { beats: :paper }
  }.freeze

  def initialize(player1, player2)
    @player1, @player2 = player1, player2
  end

  def winner
    return nil if same_pick?

    if PAIRS[@player1.pick][:beats] == @player2.pick
      @player1
    else
      @player2
    end
  end

  private

  def same_pick?
    @player1.pick == @player2.pick
  end

end

class Player

  attr_reader :pick, :name

  def initialize(name)
    @name = name
  end

  def picks(pick)
    @pick = pick
  end

end

File: javascript/lib/RockPaperScissors.js
function Player(name) {
  this.name = name;
}

Player.prototype.picks = function(pick) {
  this.pick = pick;
}

function Game(player1, player2) {
  this.player1 = player1;
  this.player2 = player2;
}

Game.prototype.PAIRS = {
			rock: 		['scissors', 'lizard'],
			scissors: ['paper', 'lizard'],
			paper: 		['rock', 'spock'],
			lizard:   ['paper', 'spock'],
			spock: 		['scissors', 'rock']
};



Game.prototype.winner = function() {
	
	if(this._isSamePick()) return null; 

 	if (this.PAIRS[this.player1.pick].indexOf(this.player2.pick) != -1 ){
 		this.loser = this.player2;
 		return this.player1;
 }
 	else {
 		this.loser = this.player1;
 		return this.player2;
 }
}

Game.prototype._isSamePick = function() {
	return this.player1.pick === this.player2.pick;
}

Game.prototype.MESSAGES = {
		rock: 		['scissors', 'lizard', 'breaks', 'crushes'],
		scissors: ['paper', 'lizard', 'cut', 'decapitate'],
		paper: 		['rock', 'spock', 'covers', 'disproves'],
		lizard:   ['paper', 'spock', 'eats', 'poisons'],
		spock: 		['scissors', 'rock', 'smashes', 'vaporizes']

};


Game.prototype.winnerMessage = function() {
	var verbIndex = this.MESSAGES[this.winner().pick].indexOf(this.loser.pick) + 2;
	return this.winner().pick + ' ' + this.MESSAGES[this.winner().pick][verbIndex] + ' ' + this.loser.pick;



	// return 'Rock breaks scissors';
};